mirrord/cli/src/config.rs

```diff
@@ -778,25 +779,83 @@ impl std::fmt::Display for ContainerRuntime {
 }
 
 #[derive(Args, Debug)]
+#[clap(args_conflicts_with_subcommands = true)]
 /// Args for the `mirrord container` command.
 pub(super) struct ContainerArgs {
+    #[clap(subcommand)]
+    pub command: Option<ContainerCommand>,
+
     #[clap(flatten)]
     /// Parameters to be passed to mirrord.
-    pub params: ExecParams,
+    pub params: Box<ExecParams>,
 
     /// Container command to be executed
     #[arg(trailing_var_arg = true)]
     pub exec: Vec<String>,
 }
 
 impl ContainerArgs {
-    pub fn into_parts(self) -> (RuntimeArgs, ExecParams) {
-        let ContainerArgs { params, exec } = self;
+    pub fn into_exec_parts(
+        self,
+    ) -> Either<(RuntimeArgs, ExecParams), (Option<PathBuf>, Option<String>)> {
+        let ContainerArgs {
+            command,
+            params,
+            exec,
+        } = self;
+
+        match command {
+            Some(command) => command.into_exec_parts(),
+            None => {
+                let runtime_args = RuntimeArgs::parse_from(
+                    std::iter::once("mirrord container --".into()).chain(exec),
+                );
```

Comment:
I coudn't make clap happy with`RuntimeArgs` straight up with `trailing_var_arg  = true` included (probably because of subcommand another `trailing_var_arg  = true` inside)